proto/sync/sync.proto

```diff
@@ -32,10 +43,92 @@ message ChangeProofRequest {
   uint32 bytes_limit = 6;
 }
 
-// KeyValue represents a single key and its value.
-message KeyValue {
+message GetChangeProofResponse {
+  ChangeProof proof = 1;
+}
+
+message VerifyChangeProofRequest {
+  ChangeProof proof = 1;
+}
+
+message VerifyChangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+message CommitChangeProofRequest {
+  ChangeProof proof = 1;
+}
+
+message CommitChangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+message GetProofRequest {
   bytes key = 1;
-  bytes value = 2;
+}
+
+message GetProofResponse {
+  Proof proof = 1;
+}
+
+// A GetRangeProofRequest requests the key-value pairs in a given key range
+// at a specific revision.
+message GetRangeProofRequest {
+  bytes root_hash = 1;
+  bytes start_key = 2;
+  bytes end_key = 3;
+  uint32 key_limit = 4;
+  uint32 bytes_limit = 5;
+}
+
+message GetRangeProofResponse {
+  RangeProof proof = 1;
+}
+
+message VerifyRangeProofRequest {
+  RangeProof proof = 1;
+}
+
+message VerifyRangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+message CommitRangeProofRequest {
+  bytes start_key = 1;
+  RangeProof range_proof = 2;
+}
+
+message CommitRangeProofResponse {
+  rpcdb.Error error = 1;
+}
+
+// ChangeProof consists of a proof of the smallest changed key,
+// a proof of the highest changed key, and the keys that have changed
+// between those. Some keys may be deleted (hence the use of KeyChange
+// instead of KeyValue).
+message ChangeProof {
+  bool had_roots_in_history = 1; // TODO remove
+  Proof start_proof = 2;
+  Proof end_proof = 3;
+  repeated KeyChange key_changes = 4;
+}
+
+// RangeProof is the response to a GetRangeProofRequest.
```

Comment:
should we update this comment?

Reply:
Thoughts on just removing all these comments? I think they're kind of redundant since this stuff is documented on `SyncableDB`

Reply:
removed